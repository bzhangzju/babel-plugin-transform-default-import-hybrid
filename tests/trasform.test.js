const {default: pluginTester} = require('babel-plugin-tester');
const plugin = require('../src/index.js');

pluginTester({
    plugin,
    tests: {
        'skip with no lib': 'import df from "./module.js";',
        'skip with named export': 'import { named } from "module";',
        'skip with namespace export': 'import * as namespace from "module";',
        'skip with default & named export': 'import df, { named } from "module";',
        'changes default import': {
            code: 'import df from "module";',
            snapshot: true,
        },
        'changes default with namespace': {
            code: 'import df, * as namespace from "module";',
            snapshot: true,
        },
        'changes default import with custom options': {
            code: 'import df from "module";',
            snapshot: true,
            pluginOptions: {
                default: 'DF',
                namespace: 'NS',
            },
        },
        'changes default with namespace and custom options': {
            code: 'import df, * as namespace from "module";',
            snapshot: true,
            pluginOptions: {
                default: 'DF',
                namespace: 'NS',
            },
        },
    },
});