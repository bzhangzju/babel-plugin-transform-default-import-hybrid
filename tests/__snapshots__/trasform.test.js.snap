// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform default import changes default import with custom options: changes default import with custom options 1`] = `

import df from "module";

      ↓ ↓ ↓ ↓ ↓ ↓

import dfDF, * as dfNS from "module";
const df = dfDF || dfNS;


`;

exports[`transform default import changes default import: changes default import 1`] = `

import df from "module";

      ↓ ↓ ↓ ↓ ↓ ↓

import df__DEFAULT__, * as df__NAMESPACE__ from "module";
const df = df__DEFAULT__ || df__NAMESPACE__;


`;

exports[`transform default import changes default with namespace and custom options: changes default with namespace and custom options 1`] = `

import df, * as namespace from "module";

      ↓ ↓ ↓ ↓ ↓ ↓

import dfDF, * as namespace from "module";
const df = dfDF || namespace;


`;

exports[`transform default import changes default with namespace: changes default with namespace 1`] = `

import df, * as namespace from "module";

      ↓ ↓ ↓ ↓ ↓ ↓

import df__DEFAULT__, * as namespace from "module";
const df = df__DEFAULT__ || namespace;


`;
